packages/app/src/cli/services/context.ts

```diff
@@ -538,3 +566,66 @@ async function logMetadataForLoadedDeployContext(env: DeployContextOutput) {
     api_key: env.identifiers.app,
   }))
 }
+
+interface DeveloperPreviewTasksContext {
+  app: OrganizationApp
+  token: string
+  enabled: boolean
+  logger: (content: OutputMessage, logger?: Logger) => void
+}
+
+export interface DeveloperPreviewInput {
+  app: OrganizationApp
+  token: string
+  hiddenOutput?: boolean
+}
+
+export async function enableDeveloperPreview(input: DeveloperPreviewInput) {
+  return developerPreviewUpdate(input, true)
+}
+
+export async function disableDeveloperPreview(input: DeveloperPreviewInput) {
+  return developerPreviewUpdate(input, false)
+}
+
+async function developerPreviewUpdate({app, token, hiddenOutput = false}: DeveloperPreviewInput, enabled: boolean) {
+  if (!app.betas?.unifiedAppDeployment) return
+
+  if (hiddenOutput) {
+    await executeDeveloperPreviewUpdate({app, token, enabled, logger: outputDebug})
+    outputDebug(`Developer preview ${enabled ? 'enabled' : 'disabled'}`)
+  } else {
+    const tasks: Task<DeveloperPreviewTasksContext>[] = [
+      {
+        title: `${enabled ? 'Enabling' : 'Disabling'} developer preview...`,
+        task: executeDeveloperPreviewUpdate,
+      },
+    ]
+
+    await renderTasks(tasks, {}, {app, token, enabled, logger: outputWarn})
+    outputCompleted(`Developer preview ${enabled ? 'enabled' : 'disabled'}`)
+    outputNewline()
+  }
+}
+
+async function executeDeveloperPreviewUpdate({app, token, enabled, logger}: DeveloperPreviewTasksContext) {
+  const query = DevelopmentStorePreviewUpdateQuery
+  const variables: DevelopmentStorePreviewUpdateInput = {
+    input: {
+      apiKey: app.apiKey,
+      enabled,
+    },
+  }
+  const result: DevelopmentStorePreviewUpdateSchema = await partnersRequest(query, token, variables)
+
+  if (
+    result.developmentStorePreviewUpdate.userErrors?.length > 0 ||
+    result.developmentStorePreviewUpdate.app.developmentStorePreviewEnabled !== enabled
+  ) {
+    logger(
```

Comment:
This feels like it's more significant than `outputDebug` level. If it worked yesterday and it fails today with no loud error message, that feels like it'll be a bad user experience.

Reply:
do you mean that the API has failed? or that the CLI try to enable it and it was already `enabled` remotely?  (same applies for disable). If the latest maybe we could remove the log message because it's not an error itself